.travis.yml

```diff
@@ -8,32 +8,38 @@ matrix:
   include:
     - os: linux
       compiler: gcc
-      env: BUILD_TYPE=Debug COVERALLS=ON COMPILER=GCC
+      env: BUILD_TYPE=Debug COVERALLS=ON COMPILER=GCC LLVM_VERSION=3.6
     - os: linux
       compiler: gcc
-      env: BUILD_TYPE=Release COVERALLS=OFF COMPILER=GCC
+      env: BUILD_TYPE=Release COVERALLS=OFF COMPILER=GCC LLVM_VERSION=3.6
     - os: linux
       compiler: clang
-      env: BUILD_TYPE=Debug COVERALLS=OFF COMPILER=CLANG
+      env: BUILD_TYPE=Debug COVERALLS=OFF COMPILER=CLANG LLVM_VERSION=3.6
     - os: linux
       compiler: clang
-      env: BUILD_TYPE=Release COVERALLS=OFF COMPILER=CLANG
+      env: BUILD_TYPE=Release COVERALLS=OFF COMPILER=CLANG LLVM_VERSION=3.6
     - os: osx
       compiler: clang
-      env: BUILD_TYPE=Debug COVERALLS=OFF COMPILER=CLANG
+      env: BUILD_TYPE=Debug COVERALLS=OFF COMPILER=CLANG LLVM_VERSION=3.7
     - os: osx
       compiler: clang
-      env: BUILD_TYPE=Release COVERALLS=OFF COMPILER=CLANG
+      env: BUILD_TYPE=Release COVERALLS=OFF COMPILER=CLANG LLVM_VERSION=3.7
+    - os: osx
+      compiler: clang
+      env: BUILD_TYPE=Debug COVERALLS=OFF COMPILER=CLANG LLVM_VERSION=latest
+    - os: osx
+      compiler: clang
+      env: BUILD_TYPE=Release COVERALLS=OFF COMPILER=CLANG LLVM_VERSION=latest
 
 install:
-  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then 'ci/install_linux.sh'; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx"   ]; then 'ci/install_macos.sh'; fi
+  - if [ "${TRAVIS_OS_NAME}" = "linux" ]; then . ci/install_linux.sh; fi
+  - if [ "${TRAVIS_OS_NAME}" = "osx"   ]; then . ci/install_macos.sh; fi
 
 script:
   - mkdir build
   - cd build
-  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then cmake -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DLLVM_DIR="/usr/share/llvm-3.6/cmake/" ..; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx"   ]; then cmake -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DLLVM_DIR="/usr/local/Cellar/llvm@3.7/3.7.1/lib/llvm-3.7/share/llvm/cmake/" ..; fi
+  - if [ "${TRAVIS_OS_NAME}" = "linux" ]; then cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DLLVM_DIR="/usr/share/llvm-${LLVM_VERSION}/cmake/" ..; fi
+  - if [ "${TRAVIS_OS_NAME}" = "osx"   ]; then cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DLLVM_DIR="$(brew --prefix ${LLVM_PACKAGE})/lib/llvm-${LLVM_VERSION}/share/llvm/cmake/" ..; fi
```

Comment:
I don't know why but the cmake module directory of the latest llvm is different from other versions. It's:
```
$(brew --prefix ${LLVM_PACKAGE})/lib/cmake/llvm # the latest llvm
$(brew --prefix ${LLVM_PACKAGE})/lib/llvm-${LLVM_VERSION}/share/llvm/cmake/ # other llvm versions
```

We might need to export `LLVM_DIR` in `ci/install_macos.sh` as well depending on the llvm unless we tweak the location to be installed, which might not a good idea.